x-pack/plugins/canvas/public/components/embeddable_flyout/flyout.component.tsx

```diff
@@ -27,38 +27,41 @@ const strings = {
 };
 export interface Props {
   onClose: () => void;
-  onSelect: (id: string, embeddableType: string) => void;
+  onSelect: (id: string, embeddableType: string, isByValueEnabled?: boolean) => void;
   availableEmbeddables: string[];
 }
 
 export const AddEmbeddableFlyout: FC<Props> = ({ onSelect, availableEmbeddables, onClose }) => {
   const embeddablesService = useEmbeddablesService();
+  const labsService = useLabsService();
```

Comment:
I think it makes more sense to pull the labsService up to the flyout.tsx and not in this component.  It's only purpose being in this file is to pass to that callback, so why not have that callback know about it instead and then this component has less to deal with.

Reply:
I still need the lab service in `flyout.component` to check if by value is enabled when filtering out embeddable factories.

https://github.com/elastic/kibana/pull/104499/files#diff-9aab885adf620bf919f3fad7f54143c060854b29a0a780272e45701aab457821R64

line 63-64
```
const availableSavedObjects = Array.from(embeddableFactories)
    .filter((factory) => isByValueEnabled || availableEmbeddables.includes(factory.type))
```

If by-value is enabled, then this conditional always evaluates to true and doesn't end up filtering out any factories, but if it is disabled, we want to restore the original functionality and filter out for only the allowed embeddable factories.

I can move labsService up to `flyout.tsx`, but I'd end up using it in both files.

Reply:
I moved `labsService` up to `flyout.tsx` and pass `isByValueEnabled` as a prop into `flyout.component` in 5163ffd.